dbt/include/global_project/macros/schema_tests/relationships.sql

```diff
@@ -1,19 +1,17 @@
 
-{% macro test_relationships(model, to, from) %}
+{% macro test_relationships(model, to, field) %}
 
-{% set column_name = kwargs.get('column_name', kwargs.get('field')) %}
+{% set column_name = kwargs.get('column_name', kwargs.get('from')) %}
 
 select count(*)
 from (
-
     select
-        {{ from }} as id
-
-    from {{ model }}
-    where {{ from }} is not null
-      and {{ from }} not in (select {{ column_name }}
-                             from {{ to }})
-
+        model.{{ column_name }} as id
```

Comment:
I'd like to avoid namespacing these fields if possible. If an expression is provided for the `field` or `from` arguments, then this code won't work as expected. I _do_ think we'll need to account for the fact that `field` and `from` could be ambiguously named though.

One way of doing that could be:
```
select count(*)
from (
    select {{ column_name }} as id from {{ model }}
) as child
left join (
    select {{ field }} as id from {{ to }}
) as parent on parent.id = child.id
where child.id is not null
  and parent.id is null
```

I just gave this a quick check against our snowplow datasets and the explain plans for the two query structures are equivalent! 

In general, I don't think that many people are using expressions in their `relationships` tests, but I'd like to either be really forward about not supporting that pattern anymore, or just continuing to support it since it's a tiny tweak

Reply:
That sounds great to me. I will certainly defer to your SQL expertise on all this :)